seastar/src/seastar_test_guard.rs

```diff
@@ -0,0 +1,21 @@
+use std::sync::{Mutex, MutexGuard};
+
+static RUNNING_TEST_WITH_SEASTAR: Mutex<()> = Mutex::new(());
+
+pub(crate) struct RunningTestWithSeastarGuard(MutexGuard<'static, ()>);
+
+/// Acquires a global mutex for the purpose of running a test with the
+/// seastar runtime.
+///
+/// Although seastar seems to support starting and stopping the runtime
+/// multiple times in a sequence, it doesn't like (and doesn't detect)
+/// when multiple runtimes are trying to run in parallel. `cargo test` runs
+/// tests in parallel by default, so it's easy to write tests which will
+/// make seastar angry (the anger manifests itself in form of segfaults).
+///
+/// The mutex should be taken by all tests that create a seastar runtime
+/// and held until the test finishes.
+pub(crate) fn acquire_guard_for_seastar_test() -> RunningTestWithSeastarGuard {
+    let guard = RUNNING_TEST_WITH_SEASTAR.lock().unwrap();
+    RunningTestWithSeastarGuard(guard)
```

Comment:
I wonder if a panic in test will poison the mutex. We could catch the panic or something (?) I guess it's not that important.

Reply:
Yes, it will poison the mutex. If other tests try to acquire the mutex, they will get an error.

I'm not sure if it matters at the moment. If a panic occurs and tries to cross the boundary through cxx, we will get an abort. However, if we solve this problem in the future then, after a test panics, the seastar runtime will have been properly stopped. In such case it will make sense to ignore the poisoning - so I'll modify the current code to ignore the poison error.